src/lib/Libifl/int_status.c

```diff
@@ -128,38 +129,21 @@ static char *statename[] = { "Transit", "Queued", "Held", "Waiting",
 static void
 decode_states(char *string, long *count)
 {
-	char *c, *s;
-	long *d;
+	char *c;
 	int i;
+	char format[10 * MAX_STATE];
```

Comment:
It's rather better to make it a static string than to create it everytime. Also, if someone changes the order of statenames it will mess your string up.